src/messaging/ExchangeMgr.cpp

```diff
@@ -94,6 +99,10 @@ CHIP_ERROR ExchangeManager::Init(SecureSessionMgr * sessionMgr)
 
 CHIP_ERROR ExchangeManager::Shutdown()
 {
+// TODO(#6919): This doesn't compile for Android, because Android has no device layer.
+#if CONFIG_DEVICE_LAYER
+    DeviceLayer::PlatformMgr().LockChipStack();
```

Comment:
Wouldn't the real solution be to have this called in the main event loop task of the stack? Seems like the stack shutting itself down should be done from within the stack task itself, rather than being called from outside with a lock.

Reply:
The stack lock was introduced to protect CHIP event processing. Putting a chip lock here can only prevent CHIP from operating, but the problem here is the exchange is also released from the client thread.

How this change can prevent leaking?

Reply:
For IM, maybe it is ok, we just moved the logic to close the exchange in OnMessageReceived, but if the exchange closing also occurs in client thread, how this locking helps?